pos_price_to_weight/static/src/js/models.js

```diff
@@ -0,0 +1,38 @@
+///*
+//    Copyright (C) 2017-Today: La Louve (<http://www.lalouve.net/>)
+//    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+//    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+//*/
+
+
+odoo.define('pos_price_to_weight.models', function (require) {
+    "use strict";
+
+    var models = require('point_of_sale.models');
+
+    var _super_PosModel = models.PosModel.prototype;
+
+    models.PosModel = models.PosModel.extend({
+
+        scan_product: function(parsed_code) {
+            if (! (parsed_code.type === 'price_to_weight')){
+                // Normal behaviour
+                return _super_PosModel.scan_product.apply(this, [parsed_code]);
+            }
+            // Compute quantity, based on price and unit price
+            var selectedOrder = this.get_order();
+            var product = this.db.get_product_by_barcode(parsed_code.base_code);
+            if(!product){
+                return false;
+            }
+            var quantity = 0;
+            var price = parseFloat(parsed_code.value) || 0;
+            if (price !== 0 && product.price !== 0){
+                quantity = price / product.price;
+            }
+            selectedOrder.add_product(product, {quantity:  quantity, merge: false});
```

Comment:
We could have a product with a price zero, shouldn't we raise a warning in such a case?

Reply:
In that case quantity will be 0, so I think it's what you'd expect

Reply:
Indeed, this module is usefull with a system that generate barcodes, generaly with a scale and a printer. Null weight does'nt seems to be a valid use case and should be handled upwind in any case.

Reply:
Works for me! 

Thank you for the clarification and migration work.